src/Microsoft.Framework.CodeGeneration.Templating/Compilation/RoslynCompilationService.cs

```diff
@@ -59,82 +36,5 @@ public CompilationResult Compile(string content)
                 return CompilationResult.Failed(content, result.ErrorMessages);
             }
         }
-
-        //Todo: This is using application references to compile the template,
-        //perhaps that's not right, we should use the dependencies of the caller
-        //who calls the templating API.
-        private List<MetadataReference> GetApplicationReferences()
-        {
-            var references = new List<MetadataReference>();
-
-            // Todo: When the target app references scaffolders as nuget packages rather than project references,
-            // since the target app does not actully use code from scaffolding framework, our dlls
-            // are not in app dependencies closure. However we need them for compiling the
-            // generated razor template. So we are just using the known references for now to make things work.
-            // Note that this model breaks for custom scaffolders because they may be using
-            // some other references which are not available in any of these closures.
-            // As the above comment, the right thing to do here is to use the dependency closure of
-            // the assembly which has the template.
-            var baseProjects = new string[] { _environment.ApplicationName, "Microsoft.Framework.CodeGeneration" };
```

Comment:
You seem to have removed this whole logic and I don't see an equivalent replacement somewhere (may be I am missing) ... this will break the scaffolding functionality... (it was on my to do list to add tests for templates but I was waiting till I finalize the content a little more)